service/qos/service_level_controller.cc

```diff
@@ -1180,7 +1191,10 @@ future<std::vector<cql3::description>> service_level_controller::auth_integratio
 }
 
 future<std::vector<cql3::description>> service_level_controller::describe_service_levels() {
-    SCYLLA_ASSERT(_auth_integration != nullptr);
+    if (_auth_integration == nullptr) {
+        throw std::runtime_error("Describing service levels requires that `auth_integration` has been registered, "
+                "but it has not. One of the potential reasons is using the maintenance socket.");
+    }
```

Comment:
Is this error propagated to the user? If yes then I don't think that the error message is too useful.